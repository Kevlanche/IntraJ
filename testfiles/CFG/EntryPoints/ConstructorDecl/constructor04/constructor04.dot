digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "ConstructorDecl[12]"; "ConstructorDecl[20]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Implicit[13]"; "Modifiers[13]"; "List[15]"; "List[16]"; "Opt[17]"; "Block[18]"; "Exit[13]"; "ExprStmt[20]"; "Entry[21]"; "Implicit[21]"; "Modifiers[21]"; "List[23]"; "List[28]"; "Opt[29]"; "Block[30]"; "Exit[21]"; "ExprStmt[32]"}
{ rank = same; "List[14]"; "List[19]"; "SuperConstructorAccess[21]"; "List[22]"; "ParameterDeclaration[24]"; "List[31]"; "SuperConstructorAccess[33]"}
{ rank = same; "List[22]"; "Modifiers[25]"; "PrimitiveTypeAccess[27]"; "List[34]"}
{ rank = same; "List[26]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < constructor04 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ConstructorDecl[12]" [label="ConstructorDecl\n < constructor04 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Implicit[13]" [label="Implicit\n < Instance >", fillcolor=white   style=filled  ]
"Modifiers[13]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[14]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[16]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[17]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[18]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[19]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[13]" [label="Exit\n ", fillcolor=white   style=filled  ]
"ExprStmt[20]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"SuperConstructorAccess[21]" [label="SuperConstructorAccess\n < super >", fillcolor=white   style=filled  ]
"List[22]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ConstructorDecl[20]" [label="ConstructorDecl\n < constructor04 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[21]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Implicit[21]" [label="Implicit\n < Instance >", fillcolor=white   style=filled  ]
"Modifiers[21]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[22]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[24]" [label="ParameterDeclaration\n < t >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[25]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[27]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[28]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[29]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[30]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[31]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[21]" [label="Exit\n ", fillcolor=white   style=filled  ]
"ExprStmt[32]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"SuperConstructorAccess[33]" [label="SuperConstructorAccess\n < super >", fillcolor=white   style=filled  ]
"List[34]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"ConstructorDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"ConstructorDecl[12]" -> "Implicit[13]" [style=dashed, color=gray]
"Modifiers[13]" -> "List[14]" [style=dashed, color=gray]
"ConstructorDecl[12]" -> "Modifiers[13]" [style=dashed, color=gray]
"ConstructorDecl[12]" -> "List[15]" [style=dashed, color=gray]
"ConstructorDecl[12]" -> "List[16]" [style=dashed, color=gray]
"ConstructorDecl[12]" -> "Opt[17]" [style=dashed, color=gray]
"Block[18]" -> "List[19]" [style=dashed, color=gray]
"ConstructorDecl[12]" -> "Block[18]" [style=dashed, color=gray]
"ConstructorDecl[12]" -> "Exit[13]" [style=dashed, color=gray]
"SuperConstructorAccess[21]" -> "List[22]" [style=dashed, color=gray]
"ExprStmt[20]" -> "SuperConstructorAccess[21]" [style=dashed, color=gray]
"ConstructorDecl[12]" -> "ExprStmt[20]" [style=dashed, color=gray]
"List[11]" -> "ConstructorDecl[12]" [style=dashed, color=gray]
"ConstructorDecl[20]" -> "Entry[21]" [style=dashed, color=gray]
"ConstructorDecl[20]" -> "Implicit[21]" [style=dashed, color=gray]
"Modifiers[21]" -> "List[22]" [style=dashed, color=gray]
"ConstructorDecl[20]" -> "Modifiers[21]" [style=dashed, color=gray]
"Modifiers[25]" -> "List[26]" [style=dashed, color=gray]
"ParameterDeclaration[24]" -> "Modifiers[25]" [style=dashed, color=gray]
"ParameterDeclaration[24]" -> "PrimitiveTypeAccess[27]" [style=dashed, color=gray]
"List[23]" -> "ParameterDeclaration[24]" [style=dashed, color=gray]
"ConstructorDecl[20]" -> "List[23]" [style=dashed, color=gray]
"ConstructorDecl[20]" -> "List[28]" [style=dashed, color=gray]
"ConstructorDecl[20]" -> "Opt[29]" [style=dashed, color=gray]
"Block[30]" -> "List[31]" [style=dashed, color=gray]
"ConstructorDecl[20]" -> "Block[30]" [style=dashed, color=gray]
"ConstructorDecl[20]" -> "Exit[21]" [style=dashed, color=gray]
"SuperConstructorAccess[33]" -> "List[34]" [style=dashed, color=gray]
"ExprStmt[32]" -> "SuperConstructorAccess[33]" [style=dashed, color=gray]
"ConstructorDecl[20]" -> "ExprStmt[32]" [style=dashed, color=gray]
"List[11]" -> "ConstructorDecl[20]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[13]" -> "Implicit[13]" [color=red, constraint=false]
"Implicit[13]" -> "Exit[13]" [color=blue, constraint=false, xlabel=" "] 
"Implicit[13]" -> "SuperConstructorAccess[21]" [color=red, constraint=false]
"SuperConstructorAccess[21]" -> "Implicit[13]" [color=blue, constraint=false, xlabel=" "] 
"SuperConstructorAccess[21]" -> "Entry[13]" [color=red, constraint=false]
"Entry[13]" -> "SuperConstructorAccess[21]" [color=blue, constraint=false, xlabel=" "] 
"Exit[21]" -> "Implicit[21]" [color=red, constraint=false]
"Implicit[21]" -> "Exit[21]" [color=blue, constraint=false, xlabel=" "] 
"Implicit[21]" -> "SuperConstructorAccess[33]" [color=red, constraint=false]
"SuperConstructorAccess[33]" -> "Implicit[21]" [color=blue, constraint=false, xlabel=" "] 
"SuperConstructorAccess[33]" -> "Entry[21]" [color=red, constraint=false]
"Entry[21]" -> "SuperConstructorAccess[33]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"ConstructorDecl[12]" -> "ConstructorDecl[20]" [style=invis]
"Entry[13]" -> "Modifiers[13]" -> "List[15]" -> "List[16]" -> "Opt[17]" -> "Block[18]" -> "Exit[13]" [style=invis]
"Entry[21]" -> "Modifiers[21]" -> "List[23]" -> "List[28]" -> "Opt[29]" -> "Block[30]" -> "Exit[21]" [style=invis]
"Modifiers[25]" -> "PrimitiveTypeAccess[27]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
44", fillcolor=1]
Edges [label="Total Nr. Edges
55", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
7", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
6", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}