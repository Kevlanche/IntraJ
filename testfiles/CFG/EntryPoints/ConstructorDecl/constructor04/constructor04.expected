NodeName: Entry[13] Successors: [SuperConstructorAccess[21]]
NodeName: SuperConstructorAccess[21] Successors: [Implicit[13]]
NodeName: Implicit[13] Successors: [Exit[13]]
NodeName: Exit[13] Successors: []
NodeName: Entry[21] Successors: [SuperConstructorAccess[33]]
NodeName: SuperConstructorAccess[33] Successors: [Implicit[21]]
NodeName: Implicit[21] Successors: [Exit[21]]
NodeName: Exit[21] Successors: []

NodeName: Exit[13] Predecessor: [Implicit[13]]
NodeName: Implicit[13] Predecessor: [SuperConstructorAccess[21]]
NodeName: SuperConstructorAccess[21] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
NodeName: Exit[21] Predecessor: [Implicit[21]]
NodeName: Implicit[21] Predecessor: [SuperConstructorAccess[33]]
NodeName: SuperConstructorAccess[33] Predecessor: [Entry[21]]
NodeName: Entry[21] Predecessor: []
